client/internal/clientstate.go

```diff
@@ -67,3 +68,26 @@ func (s *ClientSyncedState) SetAgentDescription(descr *protobufs.AgentDescriptio
 
 	return nil
 }
+
+func calcHashRemoteConfigStatus(status *protobufs.RemoteConfigStatus) {
+	// Calculate and set the Hash field from the rest of the fields in the message.
+	h := sha256.New()
+	h.Write(status.LastRemoteConfigHash)
+	h.Write([]byte(status.Status.String()))
+	h.Write([]byte(status.ErrorMessage))
+	status.Hash = h.Sum(nil)
+}
+
+func (s *ClientSyncedState) SetRemoteConfigStatus(status *protobufs.RemoteConfigStatus) error {
+	if len(status.Hash) == 0 {
+		return errors.New("hash field must be set, use CalcHashRemoteConfigStatus")
```

Comment:
If we can determine that the hash is not set and we are able to compute the hash, then should this code just compute the hash?

We could do the same above in SetAgentDescription.

Also, comparing to SetAgentDescription, should this produce an errRemoteConfigStatusMissing if status is nil?